runtime/ISO_Fortran_binding.cc

```diff
@@ -193,20 +194,149 @@ int CFI_is_contiguous(const CFI_cdesc_t *descriptor) {
   return 1;
 }
 
+static inline bool IsAssumedSize(const CFI_cdesc_t *dv) {
+  return dv->rank > 0 && dv->dim[dv->rank - 1].extent == -1;
+}
+
 int CFI_section(CFI_cdesc_t *result, const CFI_cdesc_t *source,
     const CFI_index_t lower_bounds[], const CFI_index_t upper_bounds[],
     const CFI_index_t strides[]) {
-  return CFI_INVALID_DESCRIPTOR;  // TODO
+  CFI_index_t extent[CFI_MAX_RANK];
+  CFI_index_t actualStride[CFI_MAX_RANK];
+  CFI_rank_t resRank{0};
+  char *shiftedBaseAddr{static_cast<char *>(source->base_addr)};
+
+  if (source->rank == 0) {
+    return CFI_INVALID_RANK;
+  }
+  if (IsAssumedSize(source) && upper_bounds == nullptr) {
```

Comment:
Since the arguments are all pointers, do any of the rest of them need nullptr checks?

Reply:
After discussing with @klausler , I have added nullptr checks for descriptor pointers in all CFI functions that return error codes. For the one that do not, I first need to make a function to crash nicely. This function would not belong only to the ISO runtime part so I need to make it somewhere else first.